forest/cli/src/cli/info_cmd.rs

```diff
@@ -0,0 +1,429 @@
+    let NodeStatusInfo { health, .. } = node_status;
+
+    let use_color = color.coloring_enabled();
+    let uptime_duration = Local::now() - start_time;
+    let uptime = {
+        format!(
+            "{}h {}m {}s (Started at: {})",
+            uptime_duration.num_hours(),
+            uptime_duration.num_minutes(),
+            uptime_duration.num_seconds(),
+            start_time
+        )
+    };
+
+    let chain_status = {
+        let status = chain_status(node_status)?;
+        if use_color {
+            status.blue()
+        } else {
+            status.normal()
+        }
+    };
+
+    let network = if use_color {
+        network.green()
+    } else {
+        network.normal()
+    };
+
+    let health = {
+        let s = format!("{health:.2}%\n\n");
+        if use_color {
+            if *health > 85. {
+                s.green()
+            } else {
+                s.red()
+            }
+        } else {
+            s.normal()
+        }
+    };
+
+    let default_wallet_address = {
+        let addr = default_wallet_address.clone().unwrap_or("-".to_string());
+        if use_color {
+            addr.bold()
+        } else {
+            addr.normal()
+        }
+    };
+
+    Ok(NodeInfoOutput {
+        chain_status,
+        network,
+        uptime,
+        health,
+        wallet_address: default_wallet_address,
+    })
+}
+
+#[cfg(test)]
+mod tests {
+    use std::{
+        str::FromStr,
+        sync::Arc,
+        time::{Duration, SystemTime, UNIX_EPOCH},
+    };
+
+    use chrono::Local;
+    use colored::*;
+    use forest_blocks::{BlockHeader, Tipset};
+    use forest_cli_shared::logger::LoggingColor;
+    use forest_shim::{address::Address, econ::TokenAmount};
+    use fvm_shared::clock::EPOCH_DURATION_SECONDS;
+
+    use super::{fmt_info, get_node_status, NodeStatusInfo};
+    use crate::cli::info_cmd::{chain_status, SyncStatus};
+
+    fn mock_tipset_at(duration: u64) -> Arc<Tipset> {
+        let mock_header = BlockHeader::builder()
+            .miner_address(Address::from_str("f2kmbjvz7vagl2z6pfrbjoggrkjofxspp7cqtw2zy").unwrap())
+            .timestamp(duration)
+            .build()
+            .unwrap();
+        let tipset = Tipset::from(&mock_header);
+        Arc::new(tipset)
+    }
+
+    fn node_status_good() -> NodeStatusInfo {
+        super::NodeStatusInfo {
+            behind: 0,
+            health: 90.,
+            epoch: i64::MAX,
+            base_fee: TokenAmount::from_whole(1),
+            sync_status: SyncStatus::Ok,
+        }
+    }
+
+    fn node_status_bad() -> NodeStatusInfo {
+        super::NodeStatusInfo {
+            behind: 0,
+            health: 0.,
+            epoch: 0,
+            base_fee: TokenAmount::from_whole(1),
+            sync_status: SyncStatus::Behind,
+        }
+    }
+
+    #[test]
+    fn node_status_with_null_tipset() {
+        // out of sync
+        let duration_now = SystemTime::now().duration_since(UNIX_EPOCH).unwrap();
```

Comment:
Refrain from using system time in the tests. If you want to test a wide range of timestamps, write a quickcheck test.

Reply:
converted them to quickcheck tests.